File: src/main/java/io/tatum/model/response/ledger/MarketValue.java
package io.tatum.model.response.ledger;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

/**
 * The type Market value.
 */
@AllArgsConstructor
@NoArgsConstructor
@Data
@EqualsAndHashCode
@JsonIgnoreProperties(ignoreUnknown = true)
public class MarketValue {
    /**
     * Value of transaction in given base pair.
     *
     * @type {string}
     * @memberof MarketValue
     */
    private String amount;

    /**
     * Base pair.
     *
     * @type {string}
     * @memberof MarketValue
     */
    private Fiat currency;

    /**
     * Date of validity of rate in UTC.
     *
     * @type {number}
     * @memberof MarketValue
     */
    private BigDecimal sourceDate;

    /**
     * Source of base pair.
     *
     * @type {string}
     * @memberof MarketValue
     */
    private String source;

}

File: src/main/kotlin/model/response/ledger/MarketValue.kt
package model.response.ledger

/**
 * @property amount Value of transaction in given base pair.
 * @property currency Base pair.
 * @property sourceDate Date of validity of rate in UTC.
 * @property source Source of base pair.
 */
class MarketValue(
    val amount: String? = null,
    val currency: Fiat? = null,
    val sourceDate: Long? = null,
    val source: String? = null,
)
